### PR TITLE
Add a signal handler for winshellext

### LINKED ISSUE #134: explorer.exe broken after installing the thumbnail support
**Describe the bug**
After installing f3d on windows using the installer (or registering the shell extension), a bad file could segfault f3d and break the explorer completelly until unregistration of the winshellext

https://user-images.githubusercontent.com/3129530/145490946-792006de-fc46-4b64-b133-9623719c2ad5.mp4

**System Information:**
 - OS: Windos

### PR DESCRIPTION
**Pull request by mwestphal**
Fix #134

### CODE REVIEW

**Review by Meakk — APPROVED** [review]
That seems too easy ^^
Have you tested it with broken files?

**Comment by mwestphal** [comment]
I have tested with a raise(SIGSEGV), please test the actual package from the ci with your file.

File: src/winshellext/F3DThumbnailProvider.cxx [hunk 1]
```diff
@@ -93,6 +94,14 @@ HRESULT CreateBitmapFromBuffer(vtkImageData* image, HBITMAP* phbmp)
 
   return NOERROR;
 }
+
+//------------------------------------------------------------------------------
+void SignalHandler(int signum)
+{
+  std::cerr << "Interrupt signal (" << signum << ") received.\n";
```
Comment (jpouderoux) [inline src/winshellext/F3DThumbnailProvider.cxx:101]: [Doc](https://docs.microsoft.com/en-us/cpp/c-runtime-library/reference/signal?view=msvc-170) says
"Do not issue low-level or STDIO.H I/O routines (for example, printf or fread)." so you should remove this (useless anyway) output.

Comment (mwestphal) [reply src/winshellext/F3DThumbnailProvider.cxx:101]: good point

File: src/winshellext/F3DThumbnailProvider.cxx [hunk 2]
```diff
@@ -160,6 +168,15 @@ IFACEMETHODIMP F3DThumbnailProvider::GetThumbnail(UINT cx, HBITMAP* phbmp, WTS_A
   char** argv;
   ::ConvertWCommandLineToArgs(warg, argc, argv);
 
+  // Shell extension CANNOT segfault, we must catch the signal
+  // or it can completelly break explorer.exe
+  signal(SIGTERM, ::SignalHandler);
```
Comment (jpouderoux) [inline src/winshellext/F3DThumbnailProvider.cxx:173]: Shouldn't we put this in DLL initializing func or restore the default handlers at the end of this function call?

Comment (mwestphal) [reply src/winshellext/F3DThumbnailProvider.cxx:173]: > Shouldn't we put this in DLL initializing func 

we could, but the code before cant segfault unless we made a big mistake.

> restore the default handlers at the end of this function call?

Not sure we can recover the default handlers

Comment (jpouderoux) [reply src/winshellext/F3DThumbnailProvider.cxx:173]: See the doc example. The function signal returns the previous handler. Just restore it at the end.

**Comment by mwestphal** [comment]
This approach is not going to work. We need something way more robust to not break users explorer.exe.
CreateProcess seems needed, which calls into question the whole libf3d stuff.

**Pull request closed without merging**